Lib/Formatters/Styles/OutputFormatter.php

```diff
@@ -0,0 +1,301 @@
+   *
+   * @param string $option Option name.
+   *
+   * @throws Freyja\Exceptions\invalidArgument if option name is invalid.
+   */
+  public function setOption($option) {
+    try {
+      if (!isset(static::$available_options[$option]))
+        $this->invalidArgument('option', $option);
+
+      if (!in_array(static::$available_options[$option], $this->options))
+        $this->options[] = static::$available_options[$option];
+    } catch (InvalidArgumentException $e) {
+      throw $e;
+    }
+  }
+
+  /**
+   * Unset some specific style option.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param string $option Option name.
+   *
+   * @throws Freyja\Exceptions\InvalidArgumentException if option name is invalid.
+   */
+  public function unsetOption($option) {
+    try {
+      if (!isset(static::$available_options[$option]))
+        $this->invalidArgument('option', $option);
+
+      if (false !== $pos = array_search(static::$available_options[$option], $this->options))
+        unset($this->options[$pos]);
+    } catch (InvalidArgumentException $e) {
+      throw $e;
+    }
+  }
+
+  /**
+   * Set multiple style options at once.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array $options
+   */
+  public function setOptions(array $options) {
+    $this->options = array();
+
+    foreach ($options as $option)
+      $this->setOption($option);
+  }
+
+  /**
+   * Apply style to given text.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param string $text Text to style.
+   *
+   * @return string
+   */
+  public function apply($text) {
+    $set = array();
+    $unset = array();
+
+    if (!is_null($this->foreground)) {
+      $set[] = $this->foreground['set'];
+      $unset[] = $this->foreground['unset'];
+    }
+    if (!is_null($this->background)) {
+      $set[] = $this->background['set'];
+      $unset[] = $this->background['unset'];
+    }
+    if (!is_null($this->options)) {
+      foreach ($this->options as $option) {
+        $set[] = $option['set'];
+        $unset[] = $option['unset'];
+      }
+    }
+
+    if (0 === count($set))
+      return $text;
+
+    return sprintf(
+      "\033[%sm%s\033[%sm",
+      implode(';', $set),
+      $text,
+      implode(';', $unset)
+    );
+  }
+
+  /**
+   * Utility for setting foreground and background colors.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @param string $element Element to set color to. Accepts foreground, background.
+   * @param string|null $color Color name.
+   */
+  private function setColor($element, $color) {
+    if (!in_array($color, array('foreground', 'background')))
+      throw new InvalidArgumentException('Invalid element for setColor(). Accepted values: "foreground", "background"');
+
+    if (is_null($color)) {
+      $this->$element = null;
+      return;
+    }
+
+    if (!isset(static::${'available_'.$element.'_colors'}[$color]))
+      $this->invalidArgument($element.' color', $color);
```

Comment:
Nope, I catch and rethrow the exception in `setForeground()` and `setBackground()`.